src/test/logTestReporter.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as mocha from 'mocha';
+import * as fs from 'fs';
+import * as util from 'util';
+import { TestWithLogfile, getLogFileForTest as getLogPathForTest } from './logReporterUtils';
+
+class LoggingReporter extends mocha.reporters.Spec {
+  static alwaysDumpLogs = false;
+
+  constructor(runner: any) {
+    super(runner);
+
+    runner.on('pass', (test: TestWithLogfile) => {
+      if (LoggingReporter.alwaysDumpLogs) {
+        return this.dumpLogs(test);
+      }
+    });
+
+    runner.on('fail', (test: TestWithLogfile) => {
+      return this.dumpLogs(test);
+    });
```

Comment:
huh, that's neat. I never really used the `runner` API before, this is a useful pattern